### ISSUE TITLE
Name data containing an apostrophe character is converted to &#039; in payload

### ISSUE THREAD

**Issue by jamesmacwhite** [verbb/formie#1294]
### Describe the bug

Submitting a submission that used an apostrophe in the name data was created in our Microsoft Dynamics 365 CRM with the value of: `Example O&#039;Something`, when it should have been `Example O'Something`. I've confirmed that it's something within the integration class, as bypassing and using the values directly with our CRM API does not have any issues and the apostrophe character is correctly written.

I'm assuming something in the processing of the submission and creation of the payload something is converting the character to an entity reference (possibly a sanitsation process?). The submission itself is fine as per the model data:

```php
verbb\formie\models\Name {#4483 ▼
  +prefix: null
  +firstName: "Example"
  +middleName: null
  +lastName: "O'Something"
  +name: null
  +isMultiple: true
  -_errors: null
  -_validators: null
  -_scenario: "default"
  -_events: []
  -_eventWildcards: []
  -_behaviors: []
}
```

Using `beforeSendPayload` I've confirmed that it's something being added into the payload before sending to CRM:

```php
Array
(
    [emailaddress1] => email
    [firstname] => Example
    [lastname] => O&#039;Example
)
```

### Steps to reproduce

1. Create a form using the name field
2. Populate a value using an apostrophe
3. Configure an integration to submit the data to another source
4. Observe the name value in the CRM has the entity value for the apostrophe

I don't believe it is the CRM system creating the problem, because directly pushing the payload data through the Dynamics 365 API does not replicate the issue.

### Form settings

N/A

### Craft CMS version

3.7.65.2

### Plugin version

1.6.22

### Multi-site?

No

**Comment by jamesmacwhite** [comment]
I'm wondering if this is being caused by any protection against XSS protection somewhere?

When the payload is created, the encoding appears to have been added, the model data itself is fine, so a process between the two converts it to the entity. The issue is the payload value is now the entity value and gets sent as such.

Perhaps a similar safe decode as done here? https://github.com/verbb/formie/commit/51388651cd7f795dec6ba815f0d37d7fae8ef5c8

**Comment by engram-design** [comment]
Yeah, this would probably be an XSS protection by encoding strings for safety. And as for that bit of code, that's pretty opinionated and geared towards Craft being the recipient of that content (which will handle decoded HTML). We just want to be careful about sending potentially malicious, decoded content to CRM's. `StringHelper::htmlDecode` isn't so much "safe" as the destination fields will handle rendering that content (which could be malicious). For example, Redactor will save that decoded content as encoded. And other fields will run it through the `raw` Twig filter.

I'll give it some further thought.

**Comment by jamesmacwhite** [comment]
Thank you. I appreciate the security aspect and preventing XSS is important.

For now, I think I can probably use EVENT_BEFORE_SEND_PAYLOAD to modify the value in the payload before it gets sent by parsing for HTML entities and converting them back e.g. html_entity_decode()?

**Comment by bartrylant** [comment]
Hi @jamesmacwhite 

Did you find a solution? I'm having the same problem with posting Formie values to a Craft entry. A bit annoying because my titles in the control panel are now full of &#039;
I can replace it on the front-end, but that's not a solution for the control panel.

**Comment by jamesmacwhite** [comment]
Our issue was to override the payload for the fields that is was a problem with, this was my solution

```php
private function _registerFormieBeforeSendPayload(): void
{
        Event::on(MicrosoftDynamics365::class,
	        Integration::EVENT_BEFORE_SEND_PAYLOAD,
	        static function(SendIntegrationPayloadEvent $event) {
		        $payload = $event->payload;
        
		        // Workaround for name having converted entities e.g. Person O'Something
		        // https://github.com/verbb/formie/issues/1294
		        if (array_key_exists('firstname', $payload) && array_key_exists('lastname', $payload)) {
			        $event->payload['firstname'] = StringHelper::htmlDecode($event->payload['firstname'], ENT_QUOTES);
			        $event->payload['lastname'] = StringHelper::htmlDecode($event->payload['lastname'], ENT_QUOTES);
		        }
	        }
        );
}
```

Basically, before the payload data gets sent to whatever integration parse the value with a decode.

This was specifically for Dynamics 365 CRM, but can work for others. I tried to limit the scope to quotes, as that's the most common pattern for name values, rather than allowing anything to be decoded to not introduce a security issue.

**Comment by jubalj** [comment]
This issue is also impacting email notifications for emails that contain an apostrophe.

I currently have user entering an email address in a formie field, and have formie trigger an email notification to the user entered email address. 

RFC 3696 allows for valid emails to contain apostrophes prior to @ symbol, however, due to this apostrophe issue email notifications to user-entered emails that contain an apostrophe fail, as the apostrophe is replaced with &#039 before being sent to mailer.

Is there an easy workaround or @engram-design can you look at officially supporting emails with apostrophes?

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-4 as 2.0.33"`.

**Comment by engram-design** [comment]
Fixed in [2.0.34](https://github.com/verbb/formie/releases/tag/2.0.34)
